### PR TITLE
Fix worker cannot shutdown due to resource close failed or heart beat check failed

### PR DESCRIPTION
**Pull request by ruanwenjun**
## Purpose of the pull request

- Use try-with-resource to avoid some resource may not close
- Add heartbeat check error threshold to avoid when check heartbeat throw exception may cause the worker cannot stop
- Close standalone server when starting failed

## Verify this pull request

This pull request is code cleanup without any test coverage.

*(or)*

This pull request is already covered by existing tests, such as *(please describe tests)*.

(or)

This change added tests and can be verified as follows:

### CODE REVIEW

File: dolphinscheduler-common/src/main/java/org/apache/dolphinscheduler/common/Constants.java [hunk 1]
```diff
@@ -50,6 +50,8 @@ private Constants() {
     public static final String REGISTRY_DOLPHINSCHEDULER_LOCK_FAILOVER_MASTERS = "/lock/failover/masters";
     public static final String REGISTRY_DOLPHINSCHEDULER_LOCK_FAILOVER_WORKERS = "/lock/failover/workers";
     public static final String REGISTRY_DOLPHINSCHEDULER_LOCK_FAILOVER_STARTUP_MASTERS = "/lock/failover/startup-masters";
+
+    public static final int REGISTRY_HEART_BEAT_ERROR_THRESHOLD = 5;
```
Comment (caishunfeng) [inline dolphinscheduler-common/src/main/java/org/apache/dolphinscheduler/common/Constants.java]: It's better to config by user.

File: dolphinscheduler-worker/src/main/java/org/apache/dolphinscheduler/server/worker/WorkerServer.java [hunk 2]
```diff
@@ -139,37 +137,24 @@ public void run() {
...
-                logger.warn("Worker server close wait error", e);
-            }
-
-            // close
-            this.workerRpcServer.close();
-            this.workerRegistryClient.unRegistry();
-            this.alertClientService.close();
-
             // kill running tasks
             this.killAllRunningTasks();
```
Comment (caishunfeng) [inline dolphinscheduler-worker/src/main/java/org/apache/dolphinscheduler/server/worker/WorkerServer.java:152]: I think the close order is changed, maybe it should close rpc server first to stop acceptting commands, WDYT?

Comment (ruanwenjun) [reply dolphinscheduler-worker/src/main/java/org/apache/dolphinscheduler/server/worker/WorkerServer.java:152]: The close order is not changed, `KillAllRunningTasks` doesn't rely on other component.

File: dolphinscheduler-common/src/main/java/org/apache/dolphinscheduler/common/Constants.java (hunk 1 above)
Comment (ruanwenjun) [reply dolphinscheduler-common/src/main/java/org/apache/dolphinscheduler/common/Constants.java]: Yes, agree with you.

**Review by caishunfeng — APPROVED** [review]
LGTM

**Pull request merged**